Improve api documentation in readme (#89)
sulu/SuluHeadlessBundle: README.md

management system in a headless way. 

To achieve this, the bundle includes a controller that allows to retrieve the 

> content of a **Sulu page as plain JSON content**. Furthermore, the bundle provides APIs for accessing other functionality
> over APIs (navigation contexts, snippet areas, ...) which normally are only available over twig extensions. Finally, the bundle includes

Reviewer: ```suggestion
content of a **Sulu page as plain JSON content**. Furthermore, the bundle provides APIs for accessing features that are 
available via Twig extensions in traditional templates such as navigation contexts and snippet areas. Finally, the bundle includes
```

rendering complex websites. This functionality is not available when serving the content of the website in a headless 
way, therefore the SuluHeadlessBundle includes controllers to **provide JSON APIs for accessing these features**.

 - [Navigation](#navigation)
 - [Search](#search)
 - [Snippet areas](#snippet-areas)

> By default the APIs are registered as portal urls and are so prefixed with your webspace url.
> This means if you have language specific urls the API will have the same e.g.:

Reviewer: ```suggestion
The APIs are registered as portal URLs and therefore their path is prefixed with the URL of the webspace.
If you have configured a language-specific URL for your webspace, the API URL will look something like this:
```

 - `https://example.org/en/api/...`

#### Navigation

`/api/navigations/{contextKey}`

| Parameter        | Type    | Default Value | Description
|------------------|---------|---------------|---------------------------------------

> | depth            | integer | `1`           | How depth the page tree is loaded.

Reviewer: ```suggestion
| depth            | integer | `1`           | Maximum depth of the navigation tree that is loaded.
```

> | flat             | boolean | `false`       | Return a flat list else a tree.

Reviewer: ```suggestion
| flat             | boolean | `false`       | Return navigation as flat list instead of tree.
```

> | excerpt          | boolean | `false`       | Returns also the excerpt data.

Reviewer: ```suggestion
| excerpt          | boolean | `false`       | Include excerpt data in the returned navigation.
```

Example: `/api/navigations/main?depth=2&flat=false&excerpt=true`

#### Search

`/api/search?q={searchTerm}`

| Parameter        | Type    | Default Value | Description
|------------------|---------|---------------|---------------------------------------
| q                | string  |               | The text you want to search for.

Example: `/api/search?q=CMS`

#### Snippet Areas

`/api/snippet-areas/{area}`

| Parameter        | Type    | Default Value | Description
|------------------|---------|---------------|---------------------------------------------------

> | includeExtension | boolean | `false`       | Returns also the snippet extensions like excerpt.

Reviewer: ```suggestion
| includeExtension | boolean | `false`       | Include extension data (e.g. excerpt) in the returned result.
```

rendering complex websites. This functionality is not available when serving the content of the website in a headless 
way, therefore the SuluHeadlessBundle includes controllers to **provide JSON APIs for accessing these features**.

>  - [Navigation](#navigation)
>  - [Search](#search)
>  - [Snippet areas](#snippet-areas)
> 

Reviewer: ```suggestion
```

think i would not maintain a second list here, the sections below should hopefully be enough. one thing less to think about 🙂
